course/src/isometry.cc

```diff
@@ -0,0 +1,305 @@
+            throw "Error. Invalid row index for Matrix3";
+    }
+}
+
+const Vector3& Matrix3::operator [] (const uint8_t index) const {
+    switch(index) {
+        case 0:
+            return r1_;
+        case 1:
+            return r2_;
+        case 2:
+            return r3_;
+        default:
+            throw "Error. Invalid row index for Matrix3";
+    }
+}
+
+Matrix3 Matrix3::operator + (const Matrix3& m) const {
+    return Matrix3(r1_ + m.r1_, r2_ + m.r2_, r3_ + m.r3_);
+}
+
+Matrix3 Matrix3::operator - (const Matrix3& m) const {
+    return Matrix3(r1_ - m.r1_, r2_ - m.r2_, r3_ - m.r3_);
+}
+
+Matrix3 Matrix3::operator * (const Matrix3& m) const {
+    return Matrix3(r1_ * m.r1_, r2_ * m.r2_, r3_ * m.r3_);
+}
+
+Matrix3 Matrix3::operator * (const double d) const {
+    return Matrix3(r1_ * d, r2_ * d, r3_ * d);
+}
+
+Matrix3 Matrix3::operator / (const Matrix3& m) const {
+    return Matrix3(r1_ / m.r1_, r2_ / m.r2_, r3_ / m.r3_);
+}
+
+Matrix3& Matrix3::operator = (const Matrix3& m) {
+    r1_ = m.r1_;
+    r2_ = m.r2_;
+    r3_ = m.r3_;
+    return *this;
+}
+
+Matrix3& Matrix3::operator += (const Matrix3& m) {
+    r1_ += m.r1_;
+    r2_ += m.r2_;
+    r3_ += m.r3_;
+    return *this;
+}
+
+Matrix3& Matrix3::operator -= (const Matrix3& m) {
+    r1_ -= m.r1_;
+    r2_ -= m.r2_;
+    r3_ -= m.r3_;
+    return *this;
+}
+
+Matrix3& Matrix3::operator *= (const Matrix3& m) {
+    r1_ *= m.r1_;
+    r2_ *= m.r2_;
+    r3_ *= m.r3_;
+    return *this;
+}
+
+Matrix3& Matrix3::operator /= (const Matrix3& m) {
+    r1_ /= m.r1_;
+    r2_ /= m.r2_;
+    r3_ /= m.r3_;
+    return *this;
+}
+
+bool Matrix3::operator == (const Matrix3& m) {
+    if ((r1_ == m.r1_) && (r2_ == m.r2_) && (r3_ == m.r3_)) {
+        return true;
+    }
+    return false;
+}
+
+bool Matrix3::operator != (const Matrix3& m) {
+    return !(*this == m);
+}
+
+Vector3 Matrix3::operator * (const Vector3& v) {
+    return Vector3(r1_.dot(v), r2_.dot(v), r3_.dot(v));
+}
+
+Matrix3& Matrix3::operator = (Matrix3&& m) {
+    r1_ = std::move(m.r1_);
+    r2_ = std::move(m.r2_);
+    r3_ = std::move(m.r3_);
+
+    return *this; // What about the original matrix? (m) It shouldn't have its rows available, if I'm not mistaken
+}
+
+// Getters
+Vector3 Matrix3::row(uint8_t index) const {
+    return Vector3((*this)[index]);
+}
+
+Vector3 Matrix3::col(uint8_t index) const {
+    if(index > 2) {
+        throw "Error. Invalid column index for Matrix3";
+    }
+    return Vector3(r1_[index], r2_[index], r3_[index]);
+}
+
+// Computations
+double Matrix3::det() const {
+    double subdet1 = r2_[1] * r3_[2] - r2_[2] * r3_[1];
+    double subdet2 = r2_[0] * r3_[2] - r2_[2] * r3_[0];
+    double subdet3 = r2_[0] * r3_[1] - r2_[1] * r3_[0];
+
+    return r1_[0] * subdet1 - r1_[1] * subdet2 + r1_[2] * subdet3;
+}
+
+// Class constants
+const Matrix3 Matrix3::kIdentity = 
+    Matrix3(Vector3(1,0,0), Vector3(0,1,0), Vector3(0,0,1));
+
+const Matrix3 Matrix3::kZero = Matrix3();
```

Comment:
This will work because of how you defined the default constructor of `Vector3`, but I would be more explicit here and do:
```
const Matrix3 Matrix3::kZero = Matrix3(Vector3(0,0,0), Vector3(0,0,0), Vector3(0,0,0));
```